src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -180,6 +180,76 @@ protected void AddMapInternal(string collection, string subCollecction, Indexing
             funcs.Add(map);
         }
 
+        public T? TryConvert<T>(object value)
+            where T : struct
+        {
+            if (value == null || value is DynamicNullObject)
+                return null;
```

Comment:
Dynamic null propagation will not work if this method is used. Also the values from this method will be indexed not like DynamicNullValue.ImplicitNull ones

Reply:
Yea,I don't like it, but it make sense

Reply:
Why can't we return `DynamicNullObject`?

Reply:
It will be fine. I've somehow assumed that we will have issues when passing to methods like e.g. Convert.ToInt64 when we would not recognize the dynamic type properly.